plugin/src/main/kotlin/com/github/triplet/gradle/play/PlayPublisherPlugin.kt

```diff
@@ -183,6 +183,23 @@ class PlayPublisherPlugin : Plugin<Project> {
                 generateBuildConfig.dependsOnCompat(processPackageMetadata)
             }
 
+            val publishApkTaskDependenciesHack = project.newTask(
```

Comment:
Here's the bug: when using the global publish tasks, they will be resolved before their dependencies (aka the specific publish task) which means the artifact dir will have been resolved. However, this doesn't occur when using ./gradlew with the specific task directly because the artifact dir will be resolved after the task has already been initialized.

To get around this, we add a wrapper task that's in charge of lazily adding dependencies. This works because the real task gets its artifact dir resolved and then the task dependencies get initialized with access to the resolved artifact dir.